api/types/registry/registry.go

```diff
@@ -102,3 +105,15 @@ type SearchResults struct {
 	// Results is a slice containing the actual results for the search
 	Results []SearchResult `json:"results"`
 }
+
+// DistributionInspect describes the result obtained from contacting the
+// registry to retrieve image metadata
+type DistributionInspect struct {
+	// Digest is the content addressable digest for the image on the registry
+	Digest digest.Digest
+	// ID is the local image digest
+	ID digest.Digest
```

Comment:
What is `ID` used for? If we can avoid sending it, we can mitigate some of the issues above.